windows_install_build_dependencies.bat

```diff
@@ -89,19 +90,29 @@ IF /I EXIST %bash_path% (
     :install_msys
     IF /I EXIST %msys_path% (
         rmdir /S /Q %msys_path%
-        if %errorlevel% neq 0 exit /b %errorlevel%
+        if !errorlevel! neq 0 exit /b !errorlevel!
     )
     IF /I NOT EXIST %msys_install% (
-        wget -P %opt_install_dir%\ %msys_url%
-        if %errorlevel% neq 0 exit /b %errorlevel%
+        %wget_path% -P %opt_install_dir%\ %msys_url%
+        if !errorlevel! neq 0 exit /b !errorlevel!
     )
 
     start "Installing_msys" %msys_install% in --confirm-command --accept-messages --root %msys_path%
-    if %errorlevel% neq 0 exit /b %errorlevel%
-    timeout 120
-    :: Install msys hang workaround
-    taskkill /f /t /im msys2-x86_64-20240727.exe
-    if %errorlevel% neq 0 exit /b %errorlevel%
+    for /l %%i in (1,1,90) do (
+        echo Timeout iteration: %%i
+        tasklist /NH | find /i "%msys_exe%" > nul
+        if !errorlevel! neq 0 (
+            Process "%msys_exe%" finished
+            goto install_finished
+        ) else (
+            echo Process "%msys_exe%" in progress ...
+            ping 1.1.1.1 -n 1 > nul
+        )
+    )
```

Comment:
Why do we need this?

Reply:
Because msys installation hangs at the end in 90% cases. I have added this not to always wait 120 seconds, but to check whether the install finished successfully earlier as Darek suggested.